### ISSUE TITLE
aesdchar kernel driver update

### ISSUE THREAD

**Issue by acpabst** [cu-ecen-aeld/final-project-acpabst#4]
update [aesdchar](https://github.com/cu-ecen-aeld/assignments-3-and-later-acpabst) to have a clearing function such that the circular buffer does not have to be unloaded and reloaded each time the game restarts. Having to unload the driver and reload the driver when the game starts means that no other applications could safely use the driver!
Could I use flush to do this? Maybe I need to use ioctl? 
https://www.oreilly.com/library/view/linux-device-drivers/0596000081/ch03s03.html 

DoD
Note that testing this can be done locally. After loading the new driver into the kernel, the following behavior is expected:
- write to the device, then reading should return what was written
- flushing the device, then reading should return nothing!

Summary
Each of the two parts of the DoD are true
I updated the [Assigntment 3 repo](https://github.com/cu-ecen-aeld/assignments-3-and-later-acpabst/tree/16006ca6696477ef21e1954a60a3a9b349b4430b) (commit 16006ca) with the flushing functionality, and added testing infrastructure to test it. When running locally, the device continues to return what was written as normal, but when the ioctl is invoked, the device returns nothing because the data was flushed!
Additionally, I added this to the custom raspberry pi image!

**Comment by acpabst** [comment]
Done with this!
I used an additional ioctl command to implement a flush function in the aesdchar kernel driver. I could not figure out how to invoke an ioctl from the command line, so I wrote a small helper program in c to test it, and a script to run the test. 
I went an extra step and added this to my custom Linux image, and confirmed that the new ioctl worked locally and on the Raspberry Pi. 
Here is a picture of the completed test running on the Raspberry Pi!
![IMG_1213](https://github.com/cu-ecen-aeld/final-project-acpabst/assets/43686903/b283ae63-0066-43f1-a5ef-3b9022705123)
Note that this corresponds to the following commit: [f0f0b8d](https://github.com/acpabst/memorymania-yocto/tree/f0f0b8d11d0390bb17715386e61d2a6d22d65c1b)
